frontend/src/scenes/insights/utils/cleanFilters.ts

```diff
@@ -465,6 +465,7 @@ export function cleanFilters(
 
         if (filters.date_from === 'all' || isLifecycleFilter(filters)) {
             trendLikeFilter['compare'] = false
+            trendLikeFilter['compare_to'] = undefined
```

Comment:
A somewhat radical idea I have is to turn `compare` into this type: `bool | { relativePeriodStart: string }` – in which case it would be _impossible_ to have `compare_to` set while `compare` is false. But I'm not sure if it'd actually work smoothly. @thmsobrmlr What do you think, having worked out a lot of the schema of queries?

Reply:
I don't see why this wouldn't work, but I'm also not sure if this would make it more understandable. Another option would be to use a discriminating union (frontend side, don't think python supports somthing like it) `{compare:true, relativePeriodStart: null | string} | {compare:false}`

Aside: One thing we're trying to do with the new schema is grouping related properties together in a filter e.g." `breakdownFilter`. In line with that we could have a `compareFilter` here that has all options for the compare feature.